packages/indy-vdr/src/dids/IndyVdrSovDidResolver.ts

```diff
@@ -0,0 +1,99 @@
+import type { GetNymResponseData, IndyEndpointAttrib } from './didSovUtil'
+import type { DidResolutionResult, ParsedDid, DidResolver, AgentContext } from '@aries-framework/core'
+
+import { injectable } from '@aries-framework/core'
+import { GetAttribRequest, GetNymRequest } from 'indy-vdr-test-shared'
+
+import { IndyVdrError, IndyVdrNotFoundError } from '../error'
+import { IndyVdrPoolService } from '../pool/IndyVdrPoolService'
+
+import { addServicesFromEndpointsAttrib, sovDidDocumentFromDid } from './didSovUtil'
+
+@injectable()
+export class IndyVdrSovDidResolver implements DidResolver {
+  private indyVdrPoolService: IndyVdrPoolService
+
+  public constructor(indyVdrPoolService: IndyVdrPoolService) {
+    this.indyVdrPoolService = indyVdrPoolService
+  }
```

Comment:
With the 0.3.0 release the dids module now expectes instances of dids. This is also still incorrect in the `IndySdkSovDidResolver` from the indy-sdk package, but if you look at the `IndySdkSovDidResolver` from the `core` package you can see it doesn't use constructor injection anymore.

This allows the did resolvers to be configured with their own configuration when needed (e.g. check did resolver taking an `rpcUrl`.).

(note to self: we should remove the `@injectable` from the  `IndySdkSovDidResolver` in the core package, that is not needed anymore)

Reply:
True! I've updated the implementation to take `IndyVdrPoolService` instance from `AgentContext`. I think in practice it will make easier to create an instance and pass it into dids module config.

Reply:
Awesome!